packages/react/src/ActionList/Item.tsx

```diff
@@ -288,13 +288,20 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
     const selectableRoles = ['menuitemradio', 'menuitemcheckbox', 'option']
     const includeSelectionAttribute = itemSelectionAttribute && itemRole && selectableRoles.includes(itemRole)
 
+    let focusable
+
+    // if item is disabled and is of type (menuitem*, option) it should remain focusable, if inactive, apply the same rules
+    if ((disabled && !inferredItemRole) || showInactiveIndicator) {
+      focusable = true
+    }
+
     const menuItemProps = {
       onClick: clickHandler,
       onKeyPress: !buttonSemantics ? keyPressHandler : undefined,
       'aria-disabled': disabled ? true : undefined,
       'data-inactive': inactive ? true : undefined,
       'data-loading': loading && !inactive ? true : undefined,
-      tabIndex: disabled || showInactiveIndicator ? undefined : 0,
+      tabIndex: focusable ? undefined : 0,
```

Comment:
This is mainly [adding an additional conditional](https://github.com/primer/react/pull/4481/files#diff-21bd620cf266fd1b3eb8d73399ee032c406b9bf49eac6f598ad65c9389ddb515R294) and placing it in `focusable`. 

`(disabled && !inferredItemRole) || showInactiveIndicator` instead of `disabled || showInactiveIndicator`


This change will only apply [if the container is an `ActionMenu` or a `SelectPanel`](https://github.com/primer/react/blob/main/packages/react/src/ActionList/Item.tsx#L93), otherwise the functionality will remain the same. So if an item inside of an `ActionMenu` is disabled, it will receive `0`. 

Let me know if we should opt for a different name here, or if it's still confusing!